packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -337,14 +338,16 @@ class FlutterPlugin implements Plugin<Project> {
     // Add the dependencies on other plugin projects to the plugin project.
     // A plugin A can depend on plugin B. As a result, this dependency must be surfaced by
     // making the Gradle plugin project A depend on the Gradle plugin project B.
-    private void configurePluginDependencies(String pluginName, String dependencies) {
-        Project pluginProject = project.rootProject.findProject(":$pluginName")
+    private void configurePluginDependencies(Object dependencyObject) {
+        assert dependencyObject.name instanceof String
+        Project pluginProject = project.rootProject.findProject(":${dependencyObject.name}")
         if (pluginProject == null) {
             // Ignore plugins that don't have a project since most likely they don't
             // have an android/ directory.
             return
         }
-        dependencies.split(',').each { pluginDependencyName ->
+        assert dependencyObject.dependencies instanceof List
+        dependencyObject.dependencies.each { pluginDependencyName ->
             if (pluginDependencyName.empty) {
```

Comment:
Do you need to assert that the list elements are strings?

Reply:
good catch